pkg/protocol/engine/congestioncontrol/scheduler/drr/issuerqueue.go

```diff
@@ -135,8 +139,27 @@ func (q *IssuerQueue) PopFront() *blocks.Block {
 }
 
 func (q *IssuerQueue) RemoveTail() *blocks.Block {
+	var oldestSubmittedBlock *blocks.Block
```

Comment:
I didn't look in detail how the IssuerQueue works... but shouldn't we introduce some mutexes before we loop pver submitted and pop the tail etc? Or is the logic safe against such data races?

Reply:
Protected from outside by bufferMutex